### ISSUE TITLE
Logs Not Uploading On PAX L1400

### ISSUE THREAD

**Issue by tjolsen-vn** [DataDog/dd-sdk-android#1689]
> **Describe what happened**

I am using DataDog heavily at a large number of sports and entertainment venues with [VenueNext, a Shift4 Company](https://www.shift4.com/sports-entertainment). In our latest production release, we are using the `v1.19.0` Android SDK and plugin version `1.1.0`. This is working great in production. However, in development, I've upgraded to `v2.2.0` and plugin version `1.11.0`. This version seems to be breaking logs on PAX L1400 devices. Other devices we are using are still working, it is only one specific device that isn't batching any logs at all after upgrading DD versions.

I suspect it is because of [this check](https://github.com/DataDog/dd-sdk-android/blob/fe18be92cb35bd252e1b7d966e7dde274a0e3896/dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/data/upload/DataUploadRunnable.kt#L65) in the SDK code. This particular device has an external power source. So, I think that check is failing and the SDK thinks there isn't enough battery to upload, but this device is always plugged into a power source so that would be incorrect.

![image](https://github.com/DataDog/dd-sdk-android/assets/54280016/24d32851-0f56-48d1-9f2f-9a7983448511)

The reason I think this is the issue is because I had this same issue on previous versions and was told `v1.19.0` (I think with [this PR](https://github.com/DataDog/dd-sdk-android/pull/858)) would fix the issue, and it did. The battery / external power source was the culprit last time. Seems like it broke again in newer versions. 

> **Steps to reproduce the issue:**

I suspect:
Run DataDog on any device with an external power source that reports the the battery level as "Unknown" in the Android settings.

> **Describe what you expected:**

I should still be able to use DataDog and upload logs for this device. The fact that it has an external power source should not prevent this.

> **Additional context**

 - Android OS version: <kbd>Android 11</kbd>
 - Device Model: <kbd>PAX L1400</kbd>
 - Datadog SDK version: `v2.2.0`
 - This issue is going to hurt us - we can't confidently upgrade in production if we can't see logs on this device because we are moving to using the PAX L1400 much more heavily

**Comment by xgouchet** [comment]
Hi @tjolsen-vn , and thanks for opening this ticket. 

We don't have that specific PAX L1400 in our hands, would you by chance be able to add a breakpoint in the `BroadcastReceiverSystemInfoProvider` and let us know what values are being reported there? 
In theory this class which is behind the battery check has not changed between `1.19.0` and `2.2.0`, so there might be something else at play here.

**Comment by tjolsen-vn** [comment]
@xgouchet , I apologize. I jumped to conclusions here. I found out what was actually happening. It turns out, for this specific device, we handle our app launch differently than other devices. This results in the device attempting to log to DataDog before DataDog has been initialized. Not sure why this causes no logs to ever work on our end. But, if I make sure not to attempt any logs until DataDog has been initialized and a Logger has been created, it all works on all devices. 

Given this new finding, I'd say this can be closed.
